nats.go

```diff
@@ -283,6 +283,10 @@ type CustomDialer interface {
 	Dial(network, address string) (net.Conn, error)
 }
 
+type InProcessConnProvider interface {
+	Conn() (net.Conn, error)
```

Comment:
Stupid me... unless we change the server code (from your PR: https://github.com/nats-io/nats-server/pull/2360), it has to be InProcessConn(). By the way, since your server PR has been merged, I would recommend that you update this PR to add a test that checks the new feature. You would need to update the go_test.mod file to point to the main branch of the server and then do a `go mod tidy -modfile=go_test.mod`.